From 329be11e0411658e7c1d52494b0831b437219720 Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Mon, 12 Aug 2024 15:35:53 +0530
Subject: [PATCH] Add sepolicies to use default camera hal

As Google's default HAL is used, added sepolicies.

Tests done:
- Android boot
- Camera preview and capture

Tracked-On: OAM-123336
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 camera-ext/ext-camera-only/file_contexts         | 2 +-
 camera-ext/ext-camera-only/hal_camera_default.te | 4 ++++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/camera-ext/ext-camera-only/file_contexts b/camera-ext/ext-camera-only/file_contexts
index 8f5d514..e0adc36 100644
--- a/camera-ext/ext-camera-only/file_contexts
+++ b/camera-ext/ext-camera-only/file_contexts
@@ -1,2 +1,2 @@
 /dev/media[0-9]+          u:object_r:video_device:s0
-/(vendor|system/vendor)/bin/hw/android\.vendor\.hardware\.camera\.provider\-V1-external-service       u:object_r:hal_camera_default_exec:s0
+/(vendor|system/vendor)/bin/hw/android\.hardware\.camera\.provider\-V1-external-service       u:object_r:hal_camera_default_exec:s0
diff --git a/camera-ext/ext-camera-only/hal_camera_default.te b/camera-ext/ext-camera-only/hal_camera_default.te
index aca2051..67ca30f 100644
--- a/camera-ext/ext-camera-only/hal_camera_default.te
+++ b/camera-ext/ext-camera-only/hal_camera_default.te
@@ -1,10 +1,14 @@
 #============= hal_camera_default ==============
 vndbinder_use(hal_camera_default);
 
+hal_client_domain(hal_camera_default, hal_graphics_allocator)
+
 allow hal_camera_default gpu_device:chr_file rw_file_perms;
 allow hal_camera_default gpu_device:dir search;
 allow hal_camera_default hal_graphics_allocator_default_tmpfs:file { map read write };
 allow hal_camera_default hal_graphics_mapper_hwservice:hwservice_manager find;
 allow hal_camera_default hal_graphics_composer_default:fd use;
 allow hal_camera_default self:{ socket vsock_socket } { create read write listen accept bind };
+allow hal_camera_default hal_graphics_mapper_service:service_manager find;
+allow hal_camera_default hal_graphics_allocator_default:binder call;
 set_prop(hal_camera_default, vendor_camera_default_prop)
-- 
2.45.2

